src/main/scala/com/microsoft/partnercatalyst/fortis/spark/sinks/cassandra/CassandraEventSchema.scala

```diff
@@ -0,0 +1,133 @@
+      eventtime = item.details.eventtime,
+      batchid = batchid,
+      eventlangcode = item.analysis.language.orNull,
+      eventid = item.details.eventid,
+      insertiontime = new Date().getTime,
+      body = item.details.body,
+      sourceurl = item.details.sourceurl,
+      title = item.details.title)
+  }
+}
+
+object CassandraPopularPlaces {
+  def apply(item: EventBatchEntry): Seq[PopularPlaceAggregate] = {
+    for {
+      kw <- Utils.getConjunctiveTopics(Option(item.computedfeatures.keywords))
+      location <- item.computedfeatures.places
+      periodType <- Utils.getCassandraPeriodTypes
+    } yield PopularPlaceAggregate(
+      pipelinekey = item.pipelinekey,
+      centroidlat = location.centroidlat,
+      centroidlon = location.centroidlon,
+      placeid = location.placeid,
+      periodstartdate = Period(item.eventtime.getTime, periodType).startTime(),
+      periodenddate = Period(item.eventtime.getTime, periodType).endTime(),
+      periodtype = periodType.periodTypeName,
+      period = periodType.format(item.eventtime.getTime),
+      externalsourceid = item.externalsourceid,
+      mentioncount = item.computedfeatures.mentions,
+      conjunctiontopic1 = kw._1,
+      conjunctiontopic2 = kw._2,
+      conjunctiontopic3 = kw._3,
+      avgsentiment = item.computedfeatures.sentiment.neg_avg
+    )
+  }
+}
+
+object CassandraEventTagsSchema {
+  def apply(item: EventBatchEntry): Seq[EventTags] = {
+    for {
+      kw <- item.computedfeatures.keywords
+      location <- item.computedfeatures.places
+    } yield EventTags(
+      pipelinekey = item.pipelinekey,
+      centroidlat = location.centroidlat,
+      centroidlon = location.centroidlon,
+      eventid = item.eventid,
+      topic = kw.toLowerCase,
+      eventtime = item.eventtime.getTime,
+      externalsourceid = item.externalsourceid,
+      placeid = location.placeid
+    )
+  }
+}
+
+object Utils {
+  private val ConjunctiveTopicComboSize = 3
+  private val DefaultPrimaryLanguage = "en"//TODO thread the site settings primary language to getConjunctiveTopics
+
+  def getCassandraPeriodTypes: Seq[PeriodType] = {
+    Seq(PeriodType.Day, PeriodType.Hour, PeriodType.Month, PeriodType.Week, PeriodType.Year)
+  }
+
+  def getConjunctiveTopics(topicSeq: Option[Seq[String]], langcode: Option[String] = None): Seq[(String, String, String)] = {
+    topicSeq match {
+      case Some(topics) =>
+        (topics ++ Seq("", "")).toList.combinations(ConjunctiveTopicComboSize).toList.map(combo => {
+          val sortedCombo = combo.sortWith{(a, b) =>
+            Ordering.comparatorToOrdering(Collator.getInstance(new Locale(langcode.getOrElse(langcode.getOrElse(DefaultPrimaryLanguage))))).compare(a,b) < 0 && a != ""
+          }
+
+          (sortedCombo(0), sortedCombo(1), sortedCombo(2))
+        })
+      case None => Seq()
+    }
+  }
+
+  def getStringOption(str: String): Option[String] = {
+    for (v <- Option(str) if str.nonEmpty) yield v
+  }
```

Comment:
Is this method called somewhere in a file not included in this PR? (I don't see any references to it)